### PR TITLE
Toolbox Reference

### PR DESCRIPTION
**Pull request by mactsouk**

### CODE REVIEW

File: toolbox-reference_5b7bdc1d0428631d7a8a2fcc.md [hunk 1]
```diff
@@ -1,4 +1,147 @@
...
+The `sem-service` utility has no dependencies but it needs to be executed in a
+Docker environment.
+
+#### Examples
+
+The following are valid uses of `sem-service`:
+
+	sem-service start postgres:9.6
+	sem-service start redis
+	sem-service start postgres:9.6 -e POSTGRES_PASSWORD=password
```
[the remark this answers is not in the document]
Comment (mactsouk) [reply toolbox-reference_5b7bdc1d0428631d7a8a2fcc.md]: @ervinb helped me to update the page.

File: toolbox-reference_5b7bdc1d0428631d7a8a2fcc.md [hunk 2]
```diff
@@ -1,4 +1,283 @@
...
+    -rwxr-xr-x   1 mtsouk  staff  1550 Aug 30 10:58 retry.sh
+
+In the previous example, the `retry` script succeeded after three failed tries!
+
+### cache
+
+#### Description
+
+The `cache` script is used for storing files that you want to share among the
+jobs of a Semaphore 2.0 project.
```
[the remark this answers is not in the document]
Comment (darkofabijan) [reply toolbox-reference_5b7bdc1d0428631d7a8a2fcc.md]: Lets use the example that Milana provided just not to give them ideas to use `cache` for sharing files which are not really material for caching.

File: toolbox-reference_5b7bdc1d0428631d7a8a2fcc.md [hunk 3]
```diff
@@ -1,4 +1,283 @@
...
+- `SEMAPHORE_CACHE_URL`: this environment variable stores the IP address and
+    the port number of the cache server (`x.y.z.w:29920`).
+- `SEMAPHORE_CACHE_USERNAME`: this environment variable stores the username
+    that will be used for connecting to the cache server
+	(`5b956eef90cb4c91ab14bd2726bf261b`).
+- `SSH_PRIVATE_KEY_PATH`: this environment variable stores the path to the
+    SSH key that will be used for connecting to the cache server
+	(`/home/semaphore/.ssh/semaphore_cache_key`).
+
+All these three environment variables are automatically defined and initialized
```
[the remark this answers is not in the document]
Comment (mactsouk) [reply toolbox-reference_5b7bdc1d0428631d7a8a2fcc.md]: Good point!

File: toolbox-reference_5b7bdc1d0428631d7a8a2fcc.md [hunk 4]
```diff
@@ -1,4 +1,283 @@
...
+desired files and should be unique among the cache keys of the same Semaphore
+2.0 project. However, what is important about `key_value` is that you should be
+able to recover it afterwards in order to become available to all the jobs of
+the pipeline that want to use that `key_value` afterwards.
+
+The value of the `--path` parameter should be an existing directory the
+contents of which you want to store in the cache. If you use a path with more
+than two directories in it as the parameter to `--path`, `cache` will restore
+the desired contents in the last directory of the directory path that was given.
+So, for a value of `dirA/dirB`, `cache` will store the contents of `dirA/dirB`
```
[the remark this answers is not in the document]
Comment (mactsouk) [reply toolbox-reference_5b7bdc1d0428631d7a8a2fcc.md]: Another good point. Thanks!

[the remark this answers is not in the document]
Comment (mactsouk) [reply toolbox-reference_5b7bdc1d0428631d7a8a2fcc.md]: Added your examples in the Examples section.

**Pull request merged**